src/java/org/apache/cassandra/cql3/statements/SelectStatement.java

```diff
@@ -231,7 +233,8 @@ public void authorize(ClientState state) throws InvalidRequestException, Unautho
 
     public void validate(ClientState state) throws InvalidRequestException
     {
-        // Nothing to do, all validation has been done by RawStatement.prepare()
+        if (parameters.allowFiltering && !SchemaConstants.isSystemKeyspace(table.keyspace))
```

Comment:
For insight, why did we land on adding this to `SelectStatement.validate `rather than housing it in `RawStatement.prepare` next to the check for the `parameters.isDistinct` check? You and I talked about this offline but I can't recall the conclusion there.

Reply:
`prepare` and `validate` are both called one after the other [here](https://github.com/apache/cassandra/blob/f0c97132c969cc8dd028d00c5e68ada8b0b9b9c6/src/java/org/apache/cassandra/cql3/QueryProcessor.java#L420). I put the logic in validate since it would also be called when deciding if the query should run.  The logic can be placed in either function though.

Reply:
👍